src/platform/platform_win.cpp

```diff
@@ -1082,3 +1082,8 @@ std::filesystem::path mp::platform::Platform::get_root_cert_dir() const
     // Windows doesn't use `daemon_name` for the data directory (see `program_data_multipass_path`)
     return base_dir / "Multipass" / "data";
 }
+
+bool mp::platform::Platform::can_reach_gateway(const std::string& ip) const
```

Comment:
This is never used, is it? I would prefer not to have code that doesn't get exercised lying around. 

IIANM, this is useful only for QEMU and libvirt. But libvirt is going away, which leaves only QEMU. There is a facility for QEMU things that depend on linux/macOS: `QemuPlatformDetail`. Before this PR, `can_reach_gateway` was only called from something that `QemuPlatformDetail` called anyway. With this PR, it is still always called from either linux or apple specific code. 

So why not base this PR on top of Antoni's PR and try to make `can_reach_gateway` part of the QemuPlatformDetail? It should be possible to move things using it in backend_utils there as well. That would untangle a good deal of spaghetti. Brave enough to give it a shot? 😉

Reply:
You're right, good call!